lib/cloudant.js

```diff
@@ -114,5 +114,28 @@ Cloudant.prototype.getModelObjectSettings = function(mo) {
   return undefined;
 };
 
+Cloudant.prototype.ping = function(cb) {
+  debug('Cloudant.prototype.ping');
+  var self = this;
+  var driverInst = self.getDriverInst();
+  if (driverInst) {
+    driverInst.db.list(function(err, result) {
+      returnCB(err, result);
+    });
+  } else {
+    self.dataSource.once('connected', function(err) {
+      if (err) cb('ping failed');
+      self.getDriverInst().db.list(function(err, result) {
+        returnCB(err, result);
+      });
+    });
+  }
+
+  function returnCB(err, result) {
+    debug('Cloudant.prototype.ping results %j %j', err, result);
+    if (err) cb('ping failed');
```

Comment:
would `if (err) cb(new Error('ping failed'))` work better here?

Reply:
ah true, sorry I copied the old code